File: arrays/spiral-order-matrix-ii.py
class Solution:
    def generateMatrix(self, A):
        spiral = [[0]*A for i in range(A)]
        direction = ((0, 1), (1,0), (0, -1), (-1,0))

        k = 1; count = A
        i = 0; j = 0;
        while count > 0:
            spiral[i][j] = k

            for d in direction:
                x = count-1
                while x>0:
                    i += d[0]; j += d[1];
                    k += 1
                    x -= 1
                    if i==j and i < A/2: break
                    spiral[i][j] = k

            i += 1; j += 1
            count-=2
        return spiral

File: arrays/spiral-order-matrix-ii.cpp
vector<vector<int> > Solution::generateMatrix(int A) {
    vector<vector<int>> v(A, vector<int>(A));
    int c=A;
    if(c%2!=0) c++;
    c/=2;
    
    int i=0, j=1, k=0;
    while(c--){
        t=n-k-1;
        for(i=k; i<=t; i++){v[k][i]=j++;}
        for(i=k+1; i<=t; i++){v[i][t]=j++;}
        for(i=t-1; i>=k; i--){v[t][i]=j++;}
        for(i=t-1; i>k; i--){v[i][k]=j++;}
        k++;
    }
    return v;
}
